### PR TITLE
Saml missing NameIDFormat

### LINKED ISSUE #1499: SAML NPE for service metadata without NameIDFormat
Seeing an NPE when parsing service metadata that doesn't include a `NameIDFormat` tag. from the spec [1]:

```
In the absence of a <saml2p:NameIDPolicy> Format attribute in the Service Provider's
<saml2p:AuthnRequest> message, or a <md:NameIDFormat> element in the Service Provider's
metadata, the Format of the <saml2:NameID> SHOULD be set to
urn:oasis:names:tc:SAML:2.0:nameid-format:transient.
```

NPE occurs here: https://github.com/Jasig/cas/blob/master/cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/profile/builders/SamlProfileSamlSubjectBuilder.java#L47

[1] http://saml2int.org/profile/current/#section92

### PR DESCRIPTION
**Pull request by frett**
Closes #1499

### CODE REVIEW

File: cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/profile/builders/SamlProfileSamlNameIdBuilder.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 public class SamlProfileSamlNameIdBuilder extends AbstractSaml20ObjectBuilder implements SamlProfileObjectBuilder<NameID> {
     private static final long serialVersionUID = -6231886395225437320L;
 
+    private static final String DEFAULT_NAME_ID_FORMAT = "urn:oasis:names:tc:SAML:2.0:nameid-format:transient";
```
Comment (mmoayyed) [inline cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/profile/builders/SamlProfileSamlNameIdBuilder.java]: I think this should be changed to use `unspecified`. Also, I am somewhat sure opensaml has the same constant somewhere. could we see if we could use that instead?

**Comment by mmoayyed** [comment]
:+1: with a comment on constant selection.

File: cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/profile/builders/SamlProfileSamlNameIdBuilder.java (hunk 1 above)
Comment (frett) [inline cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/web/idp/profile/builders/SamlProfileSamlNameIdBuilder.java]: yeah, I'll search for it later this evening or tomorrow.

**Comment by frett** [comment]
I'm going off the behavior specified here: http://saml2int.org/profile/current/#section92

it says:
```
In the absence of a ... <md:NameIDFormat> element in the Service Provider's metadata, the Format of
the <saml2:NameID> SHOULD be set to urn:oasis:names:tc:SAML:2.0:nameid-format:transient.
```

**Comment by mmoayyed** [comment]
Excellent. Thanks for the reference. Is there a const available in opensaml we could use? Otherwise looks great

**Comment by frett** [comment]
found the constant, going to wait for travis to finish then merge.

**Comment by mmoayyed** [comment]
Just thought of something; I realize we must stick to the spec and that's perfectly fine, but is there a use case/need to externalize the default nameid format into a property, that is set to the spec expected value? I am thinking of cases where metadata is not able to provide this setting for whatever reason, and yet the SP sort of expects it. That is a bad implementation, but we could technically allow for that kind of thing. 

Not sure if it's any use a all yet, but in general, I prefer things to be configurable when the effort is negligent.

**Comment by frett** [comment]
If we do support that I think it should be defined per service and not globally for the entire server.

There is already a workaround for that use case by duplicating the service metadata and adding in the corresponding `NameIDType` to a local copy of the metadata.

**Comment by mmoayyed** [comment]
That's fine with me. I think the registry model is flexible enough now to account for options settings. 

Shall we log this as an issue for now and see how use cases pan out?

**Comment by frett** [comment]
Because it's a deviation from spec I don't think we necessarily need to log it as an issue yet, I would wait for the actual use case to present itself through usage.

**Comment by mmoayyed** [comment]
Sounds good.

**Pull request merged**